packages/react/src/analytics/integrations/Vitorino/Vitorino.ts

```diff
@@ -10,59 +10,54 @@
  * ```
  */
 
-import {
-  DATA_TEST_SELECTOR,
-  ENVIRONMENT_TYPES,
-  ERROR_MESSAGE_PREFIX,
-  GET_EVENTS_MAPPER_FN,
-  INVALID_CUSTOM_MAPPER_ERROR_MESSAGE,
-  INVALID_MAPPER_RETURN_TYPE_ERROR,
-  MARKETING_API_PREFIX,
-  MISSING_CONTEXT_ERROR_MESSAGE,
-  PROD_SCRIPT_SRC,
-  VITORINO_CALL_ERROR_MESSAGE,
-} from './constants';
 import {
   EventData,
+  IntegrationFactory,
   integrations,
   LoadIntegrationEventData,
   SetUserEventData,
   StrippedDownAnalytics,
   TrackTypesValues,
   utils,
 } from '@farfetch/blackout-analytics';
-import { postTracking } from '@farfetch/blackout-client';
-import isArray from 'lodash/isArray';
+import { Forter, Riskified } from './providers';
+import {
+  getEnvironmentFromOptions,
+  getSafeOptionsForIntegration,
+} from './vitorino-helper';
+import { VitorinoProviders } from './constants';
+import omit from 'lodash/omit';
+import type { ForterIntegrationOptions } from './providers/Forter/types';
+import type { RiskifiedIntegrationOptions } from './providers/Riskified/types';
 import type {
-  Config,
+  VitorinoIntegrationBaseOptions,
   VitorinoIntegrationOptions,
-  VitorinoTrackCallback,
-} from './types/types';
-import type { WebContextType } from '../../context';
+} from './types';
 
 /**
  * Vitorino integration.
  */
-export default class Vitorino extends integrations.Integration<VitorinoIntegrationOptions> {
-  isScriptLoaded: boolean;
-  getEventsMapper: typeof GET_EVENTS_MAPPER_FN;
-  isVitorinoConfigured: boolean;
-  vitorinoTrackCallback: VitorinoTrackCallback;
-  initializePromiseResolve: ((value?: unknown) => void) | null;
-  userIdPromiseResolve: ((value?: unknown) => void) | null;
-  initializePromise: Promise<unknown> | null;
-  userIdPromise: Promise<unknown> | null;
-  config: Config | null;
-
-  /**
-   * This integration is required, so it must be loaded independently of user
-   * consent.
-   *
-   * @returns If the integration should load or not.
-   */
-  static override shouldLoad() {
-    return true;
-  }
+class Vitorino extends integrations.Integration<VitorinoIntegrationOptions> {
+  private forter?: Forter;
```

Comment:
I think `forterInstance` would be a better name. What do you think? (Same for `riskified` prop)

Reply:
I agree and applied.